releases/v0.23.md

```diff
@@ -0,0 +1,266 @@
+# v0.23 🚀
+
+This document contains all release notes pertaining to the `v0.23.x` releases of
+Vela.
+
+## For Vela Administrators
+
+- When upgrading from `v0.23`, please take note of the [migration information](/migrations/v0.23/deploy-recovery.md)
+
```

Comment:
We need to add the other migration information regarding SQL queries that need to be ran? Maybe after https://github.com/go-vela/community/pull/906 merged?

Reply:
I referenced it and waiting for #906  to be merged